src/stores/RootStore.ts

```diff
@@ -73,9 +74,20 @@ export default class RootStore {
 		let messagesStoreState: MessagesStoreURLState = {};
 
 		if (activeWorkspace && isWorkspaceStore(activeWorkspace)) {
+			const clearEmpty = (filter: Partial<MessageFilterState> | Partial<EventsFilter>) => {
+				const tempFilter: Partial<MessageFilterState> | Partial<EventsFilter> = {};
```

Comment:
i think you can just import toJS from mobx and replace these two lines with
`
	const tempFilter = toJS(filter)
`
it will convert observable to be plain js object